src/System.Private.ServiceModel/tools/PRService/pr.ashx

```diff
@@ -182,31 +231,58 @@ public class PullRequestHandler : IHttpHandler
             string.Format("checkout {0}", branch)
         };
 
-        return RunGitCommands(gitCommands, executionResult);
+        return RunGitCommands(gitCommands, gitRepoPath, executionResult);
+    }
+
+    // Gets a list of branches 
+    // - that are available in the repo (remoteBranches == false)
+    // - that are available in defined remotes (remoteBranches == true)
+    private bool GetBranches(string gitRepoPath, bool remoteBranches, out string[] branches, StringBuilder executionResult)
+    {
+        bool success = false;
+        branches = null;
+
+        StringBuilder branchesString = new StringBuilder();
+
+        if (remoteBranches)
+        {
+            success = RunGitCommands(new string[] { "branch --remote --list" }, gitRepoPath, branchesString);
+        }
+        else
+        {
+            success = RunGitCommands(new string[] { "branch --list" }, gitRepoPath, branchesString);
+        }
+
+        executionResult.Append(branchesString);
+
+        if (success)
+        {
+            branches = branchesString.ToString().Split(new char[] { '\n', '\r', ' ', '*' }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
"origin" is done by convention - it's the immediate repo that your local repo is linked to. 
"upstream" is a convention that specifies that your immediate repo was forked from somewhere else. 

In this case, origin refers to https://github.com/dotnet/wcf, so you will always be referring to origin. For the purposes of PRService and how it is set up, you will never set up "upstream"